src/lib/builders.ts

```diff
@@ -236,6 +236,7 @@ export class ItemBuilder {
  * @returns {boolean}
  */
 const validRecipe = (recipe: GeneratorRecipe): boolean => {
+    recipe.characterSets = [...new Set(recipe.characterSets)];
```

Comment:
This is a clever solution 😄  

We should avoid changing values inside these kinds of validation functions. For example, if I didn't know what was in this function I wouldn't expect it to change anything about my recipe.

Instead of re-assigning `recipe.characterSets`, could you move `[...new Set()]` to the line where we assign the recipe:
https://github.com/1Password/connect-sdk-js/blob/e02b5da9cbc2392e7d6280a6c0923b6bc2a20d4c/src/lib/builders.ts#L133

Reply:
Sure. Sounds like a better thing to do.
Adjusted properly.